formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/ProtobufDecoding.kt

```diff
@@ -90,7 +90,10 @@ internal open class ProtobufDecoder(
             if (protoId == protoTag) return i
         }
 
-        return -1
+        throw ProtobufDecodingException(
+            "$protoTag is not among valid ${descriptor.serialName} enum indicies, " +
+                    "values size is ${desc.elementNames}"
```

Comment:
Given that it is 'values size', maybe you meant `desc.elementsCount`? Anyway, count/size is not very helpful here because proto numbers can be arbitrary

Reply:
Good spot